Documentation/git-maintenance.txt

```diff
@@ -70,6 +70,21 @@ gc::
 	be disruptive in some situations, as it deletes stale data. See
 	linkgit:git-gc[1] for more details on garbage collection in Git.
 
+loose-objects::
+	The `loose-objects` job cleans up loose objects and places them into
+	pack-files. In order to prevent race conditions with concurrent Git
+	commands, it follows a two-step process. First, it deletes any loose
+	objects that already exist in a pack-file; concurrent Git processes
+	will examine the pack-file for the object data instead of the loose
+	object. Second, it creates a new pack-file (starting with "loose-")
+	containing a batch of loose objects. The batch size is limited to 50
+	thousand objects to prevent the job from taking too long on a
+	repository with many loose objects. The `gc` task writes unreachable
+	objects as loose objects to be cleaned up by a later step only if
+	they are not re-added to a pack-file; for this reason it is not
+	advisable to enable both the `loose-objects` and `gc` tasks at the
+	same time.
+
```

Comment:
I'm not sure if you've covered this or not (or if this is the correct place to mention it).

after we collect a batch of loose objects into a new packfile, do we immediately delete those loose
objects or do we leave them as is -- and assume that the next invocation will detect that they are
now present in a packfile (probably created by the previous invocation).  Such lazy delete will help
avoid problems on Windows where we could get share-deny/file-in-use in either the maintenance
process or a concurrent Git process.

Reply:
This follows the pattern from the existing Scalar LooseObjectsStep which does a `prune-packed` followed by a `pack-objects`. When run daily, this would only trip up processes that last 24 hours (and somehow have an open handle to a loose object that whole time).